pkg/updatestrategy/aws_ec2.go

```diff
@@ -0,0 +1,212 @@
+package updatestrategy
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/aws/aws-sdk-go/service/cloudformation/cloudformationiface"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/aws/aws-sdk-go/service/elb/elbiface"
+	"github.com/zalando-incubator/cluster-lifecycle-manager/api"
+	"github.com/zalando-incubator/cluster-lifecycle-manager/pkg/spotio"
+)
+
+const (
+	spotIOOceanIDTagKey = "spot.io/ocean-id"
+)
+
+// EC2NodePoolsBackend defines a node pool consiting of externally managed EC2
+// instances
+type EC2NodePoolsBackend struct {
```

Comment:
`SpotIOBackend`? Or do you see other potential instance managers?

Reply:
The idea was to have a generic EC2 pool and then use whatever way to find the desired config. Right now I explicitly use the spotio API for this, but I wanted to abstract this part. However for now we don't have other use cases, so I can make it spot.io specific.

Reply:
Changed!